com.unity.render-pipelines.universal/Runtime/ScriptableRenderer.cs

```diff
@@ -1193,9 +1211,10 @@ void SetRenderPassAttachments(CommandBuffer cmd, ScriptableRenderPass renderPass
                 else
                 {
                     // Only setup render target if current render pass attachments are different from the active ones
-                    if (passColorAttachment != m_ActiveColorAttachments[0] || passDepthAttachment != m_ActiveDepthAttachment || finalClearFlag != ClearFlag.None)
+                    if (passColorAttachment != m_ActiveColorAttachments[0] || passDepthAttachment != m_ActiveDepthAttachment || finalClearFlag != ClearFlag.None ||
+                        renderPass.colorStoreAction != m_ActiveColorStoreActions[0] || renderPass.depthStoreAction != m_ActiveDepthStoreAction)
```

Comment:
Not for this PR, but maybe for the future we could store the current attachment setup with something like https://docs.unity3d.com/ScriptReference/Rendering.RenderTargetBinding.html then it would make code more readable to compare pass binding with active binding.

Reply:
This seems to be the only place where `m_ActiveColorStoreActions` is read, but we only read `m_ActiveColorStoreActions[0]`. Is that on purpose?

Reply:
this is the non MRT path so there should be only one color attachment (the first check that was already there only checks for color[0]:  passColorAttachment != m_ActiveColorAttachments[0])

In general this optimization PR is only for the forward renderer at the moment, so I didn't do anything in MRT/deferred branches

Reply:
regarding redundant render target switches: I suppose that could happen potentially. It doesn't happen with the current optimizations I did, but if users want to customize further their store actions in some cases it could lead to unoptimal switches I guess. But not sure there is much we can do about it. If we expose more low level customization options this is one of the risks. more power->more responsibility?

Reply:
I also noticed that in some cases the engine would still do extra target switches compared to URP SetRenderTarget calls. i.e. is opaque would Store depth, and transparent would discard.
Without the store actions comparison check: 

- Opaque SetRenderTarget is called: Opaque will store all the results
- Transparent SetRenderTarget is called: will load from Opaque, will store the all results (waste of bandwidth)

With this new check introduced the behaviour will be the same in terms of loads but now the transparent pass will discard/resolve everything, no stores